### ISSUE TITLE
Save flank output as json file

### ISSUE THREAD

**Issue by adamfilipow92** [Flank/flank#1437]
**Author the user story for this feature**

As a Flank user, I want to have the output saved to the file so I can easily parse this file on CI or use it in Flank integration tests.

**Describe the solution**

1. Add command ```--output-file-format``` to determine the output file type.
1. Save output file to matrix directory
1. Upload to gcs
1. Use it in integration tests it could help with the easiest tests update

What team think about this proposal?
@bootstraponline @piotradamczyk5 @jan-gogo @pawelpasterz @Sloox @axelzuziak-gogo

**Comment by piotradamczyk5** [comment]
I think that we should prepare some output json with information about tests results

**Comment by bootstraponline** [comment]
What output do we need that’s not already captured in junit xml?

**Comment by adamfilipow92** [comment]
I thinking about something like Flank console output but in json format. That output could contain
1. args
1. weblinks
1. test results
1. cost
1. errors if exist

**Comment by bootstraponline** [comment]
That makes sense. Maybe we should look at how gcloud CLI supports saving output to JSON and implement that?

> The default output from describe commands is YAML format, but you can use the --format flag to choose between JSON, YAML, and text output formats. JSON formatted output can be useful if you are parsing the output, while text formatted output puts each property on a separate line.
>
> https://cloud.google.com/compute/docs/gcloud-compute/tips

**Comment by adamfilipow92** [comment]
> That makes sense. Maybe we should look at how gcloud CLI supports saving output to JSON and implement that?
> 
> > The default output from describe commands is YAML format, but you can use the --format flag to choose between JSON, YAML, and text output formats. JSON formatted output can be useful if you are parsing the output, while text formatted output puts each property on a separate line.
> > https://cloud.google.com/compute/docs/gcloud-compute/tips

Yes, I will check how it's looks.

**Comment by piotradamczyk5** [comment]
# Project name

> Summary of the proposed change

# References [optional for issue commens]

> Link to supporting documentation, GitHub tickets, etc.

# Motivation

> What problem is this project solving?

# Goals

> Identify success metrics and measurable goals.

# Non-Goals [optional]

> Identify what's not in scope.

# Design

> Explain and diagram the technical design
>
> Identify risks and edge cases

# API

> What will the proposed API look like?
  
# Results

> What was the outcome of the project?

# Dependencies

> What is the project blocked on?

> What will be impacted by the project?

# Testing

> How will the project be tested?

# Alternatives Considered [optional]

> Summarize alternative designs (pros & cons)

# Timeline [optional for regular tigers]

> Document milestones and deadlines.

DONE:

  -

NEXT:

  -

**Comment by jan-gogo** [comment]
I would love to see also the proposition of structured JSON or YAML output. This could help to identify the best way for implementation.
